packages/create-cloudflare/src/common.ts

```diff
@@ -591,15 +578,25 @@ function prepareCommitMessage(commitMessage: string): string {
 }
 
 export function quoteShellArgs(args: string[]): string {
-	if (process.platform !== "win32") {
-		return quote(args);
-	} else {
+	if (process.platform === "win32") {
 		// Simple Windows command prompt quoting if there are special characters.
 		const specialCharsMatcher = /[&<>[\]|{}^=;!'+,`~\s]/;
 		return args
 			.map((arg) =>
 				arg.match(specialCharsMatcher) ? `"${arg.replaceAll(`"`, `""`)}"` : arg
 			)
 			.join(" ");
+	} else {
+		return args
+			.map((s) => {
+				if (/["\s]/.test(s) && !/'/.test(s)) {
+					return "'" + s.replace(/(['\\])/g, "\\$1") + "'";
+				}
+				if (/["'\s]/.test(s)) {
+					return '"' + s.replace(/(["\\$`!])/g, "\\$1") + '"';
```

Comment:
Chatted offline but rather than mess with the implementation here which is copied directly from `shell-quote`, we'll leave it as is (since it improves the sitatuion) and pursue creating our own thing in a separate PR.